interview-questions.md

```diff
@@ -5,46 +5,65 @@ First, without external resources. Challenge yourself to answer from memory.
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
+As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+Your answer: The foreign key would be on the Student model, the cohort has many students.
 
-  Researched answer:
+Researched answer:
 
+The foreign key is used to identify which cohort the student belongs to- it is not the same as the student id but specifically identifies which cohort the student is in.
 
+Which RESTful routes must always be passed params? Why?
 
-2. Which RESTful routes must always be passed params? Why?
+Your answer: The Restful routes must always passed params because it defines the actions available in the applications and the type of action such as get, post, and patch.
 
-  Your answer:
+Researched answer:
+Route parameters are named URL segments that are used to capture the values specified at their position in the URL. The captured values are populated in the req.params object, with the name of the route parameter specified in the path as
+their respective keys.
 
-  Researched answer:
 
+Name three rails generator commands. What is created by each?
 
+Your answer:  Rails Generate will generate a new code, rails s: will start the application model, rails c will generate a new code (new input).
```

Comment:
You are right that `Rails generate` will create new code. But what to the specific generate commands create?